openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIRequestBodyMapper.java

```diff
@@ -243,28 +243,30 @@ private void handleMultipleMIMETypes(RequestBody bodyParameter,
                                          RequiredParameterNode payloadNode, Map<String, Schema> schema) {
 
         for (MappingFieldNode fieldNode: fields) {
-            if (((SpecificFieldNode) fieldNode).fieldName().toString().equals(MEDIA_TYPE)) {
-                if (fieldNode.children() != null) {
-                    RequestBody requestBody = new RequestBody();
-                    SpecificFieldNode field = (SpecificFieldNode) fieldNode;
-                    if (field.valueExpr().isPresent()) {
-                        ExpressionNode valueNode = field.valueExpr().get();
-                        if (valueNode instanceof ListConstructorExpressionNode) {
-                            SeparatedNodeList mimeList = ((ListConstructorExpressionNode) valueNode).expressions();
-                            if (mimeList.size() != 0) {
-                                for (Object mime : mimeList) {
-                                    if (mime instanceof BasicLiteralNode) {
-                                        createRequestBody(bodyParameter, payloadNode, schema, requestBody,
-                                                ((BasicLiteralNode) mime).literalToken().text());
-                                    }
-                                }
-                            }
-                        } else {
+            if (!(fieldNode instanceof SpecificFieldNode) ||
+                    !((SpecificFieldNode) fieldNode).fieldName().toString().equals(MEDIA_TYPE) ||
+                    fieldNode.children() == null) {
+                continue;
+            }
+            RequestBody requestBody = new RequestBody();
+            SpecificFieldNode field = (SpecificFieldNode) fieldNode;
+            if (field.valueExpr().isEmpty()) {
+                continue;
+            }
+            ExpressionNode valueNode = field.valueExpr().get();
+            if (valueNode instanceof ListConstructorExpressionNode) {
+                SeparatedNodeList mimeList = ((ListConstructorExpressionNode) valueNode).expressions();
+                if (mimeList.size() != 0) {
```

Comment:
I think this check is redundant since foreach loop can handle empty lists by itself